clients/include/testing_scal_batched.hpp

```diff
@@ -0,0 +1,223 @@
+ * Copyright 2018-2019 Advanced Micro Devices, Inc.
+ * ************************************************************************ */
+
+#include "cblas_interface.hpp"
+#include "flops.hpp"
+#include "norm.hpp"
+#include "rocblas.hpp"
+#include "rocblas_init.hpp"
+#include "rocblas_math.hpp"
+#include "rocblas_random.hpp"
+#include "rocblas_test.hpp"
+#include "rocblas_vector.hpp"
+#include "unit.hpp"
+#include "utility.hpp"
+
+template <typename T, typename U = T>
+void testing_scal_batched(const Arguments& arg)
+{
+    rocblas_int N           = arg.N;
+    rocblas_int incx        = arg.incx;
+    U           h_alpha     = arg.get_alpha<U>();
+    rocblas_int batch_count = arg.batch_count;
+
+    rocblas_local_handle handle;
+
+    // argument sanity check before allocating invalid memory
+    if(N < 0 || incx <= 0 || batch_count <= 0)
+    {
+        device_vector<T*, 0, T> dx(1);
+        if(!dx)
+        {
+            CHECK_HIP_ERROR(hipErrorOutOfMemory);
+            return;
+        }
+
+        CHECK_ROCBLAS_ERROR(rocblas_set_pointer_mode(handle, rocblas_pointer_mode_host));
+        if(batch_count < 0)
+            EXPECT_ROCBLAS_STATUS(
+                (rocblas_scal_batched<T, U>)(handle, N, &h_alpha, dx, incx, batch_count),
+                rocblas_status_invalid_size);
+        else
+            CHECK_ROCBLAS_ERROR(
+                (rocblas_scal_batched<T, U>)(handle, N, &h_alpha, dx, incx, batch_count));
+        return;
+    }
+
+    size_t size_x = N * size_t(incx);
+
+    // Naming: dX is in GPU (device) memory. hK is in CPU (host) memory, plz follow this practice
+
+    // Device-arrays of pointers to device memory
+    device_vector<T*, 0, T> dx_1(batch_count);
+    device_vector<T*, 0, T> dx_2(batch_count);
+    device_vector<U>        d_alpha(1);
+    if(!dx_1 || !dx_2 || !d_alpha)
+    {
+        CHECK_HIP_ERROR(hipErrorOutOfMemory);
+        return;
+    }
+
+    // Host-arrays of pointers to host memory
+    host_vector<T> hx_1[batch_count];
+    host_vector<T> hx_2[batch_count];
+    host_vector<T> hx_gold[batch_count];
+
+    // Host-arrays of pointers to device memory
+    // (intermediate arrays used for the transfers)
+    device_batch_vector<T> x_1(batch_count, size_x);
+    device_batch_vector<T> x_2(batch_count, size_x);
+
+    for(int i = 0; i < batch_count; i++)
+    {
+        hx_1[i]    = host_vector<T>(size_x);
+        hx_2[i]    = host_vector<T>(size_x);
+        hx_gold[i] = host_vector<T>(size_x);
+    }
+
+    int last = batch_count - 1;
+    if((!x_1[last] && size_x) || (!x_2[last] && size_x))
+    {
+        CHECK_HIP_ERROR(hipErrorOutOfMemory);
+        return;
+    }
+
+    // Initial Data on CPU
+    rocblas_seedrand();
+    for(int i = 0; i < batch_count; i++)
+    {
+        rocblas_init<T>(hx_1[i], 1, N, incx);
+
+        hx_2[i]    = hx_1[i];
+        hx_gold[i] = hx_1[i];
+    }
+
+    // copy data from CPU to device, does not work for incx != 1
```

Comment:
Is the "does not work" still true?

Reply:
This comment seems to be everywhere in the blas1 testing files. I'm not sure why it would not work for incx != 1 as, in this case, size_x includes the incx value. Also, we have tests which test for values besides incx == 1. I'm guessing this has been fixed/can be removed but I'm unsure of the original reasoning.

Reply:
I'm guessing it was left over from before my mass-refactoring of `rocblas-test`.

If it works now, and passes tests for which the comment contradicts, we can remove the comment.